### PR TITLE
Adding optional read_only connection string that can enable connectio…

### PR DESCRIPTION
**Pull request by kirutthika**
…n to read replica for GetNodeSelectors and ListRegistrationEntries operations.

Signed-off-by: Kirutthika <kirutthika.raja@uber.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
Adding read_only database connection string that can be used for read only replicas

**Description of change**
To take advantage of read replicas, enabling MySQL plugin to allow configuration of a read-only connection string.
Write connection string is mandatory, and if no Read-only is set then the Write string is used for Read as well. If Read is set, then the read connection string is use for Read operations, more specifically for GetNodeSelectors and ListRegistrationEntries.

**Which issue this PR fixes**
#1239

### CODE REVIEW

File: conf/server/server.conf [hunk 1]
```diff
@@ -19,6 +19,7 @@ plugins {
         plugin_data {
             database_type = "sqlite3"
             connection_string = "./.data/datastore.sqlite3"
+            ro_connection_string = "./.data/datastore.sqlite3"
```
[the remark this answers is not in the document]
Comment (kirutthika) [reply conf/server/server.conf]: Done.

File: doc/plugin_server_datastore_sql.md [hunk 2]
```diff
@@ -51,6 +53,9 @@ The `connection_string` for the PostreSQL database connection consists of the nu
 ```
 connection_string="dbname=postgres user=postgres password=password host=localhost sslmode=disable"
 ```
+```
```
[the remark this answers is not in the document]
Comment (kirutthika) [reply doc/plugin_server_datastore_sql.md]: Done.

File: pkg/server/plugin/datastore/sql/sql.go [hunk 3]
```diff
@@ -349,6 +351,9 @@ func (ds *Plugin) GetNodeSelectors(ctx context.Context,
 	callCounter := ds_telemetry.StartGetNodeSelectorsCall(ds.prepareMetricsForCall())
 	defer callCounter.Done(&err)
 
+	if ds.roDb != nil {
```
[the remark this answers is not in the document]
Comment (kirutthika) [reply pkg/server/plugin/datastore/sql/sql.go]: Added a new boolean field to the rpc request.

**Comment by kirutthika** [comment]
> Thanks for preparing this @kirutthika! This is exciting. Along with a few smaller comments, I have two bigger pieces of feedback:
> 
> 1. Use of the read-only connection should be opt-in for the RPCs that support it (e.g. `bool tolerate_stale`). This will require updating datastore.proto and factoring that boolean into the decision to use `roDB` instead of `db`.
> 2. The read-only connection doesn't make sense for SQLite3 and IMO should be a configuration error.
> 
> That second point makes unit-testing this impractical since the unit-test suite runs SQLite3 by default. One option is to revert the current changes to the unit tests and author new two standalone tests that setup a database with a read-only connection string and do a ListRegistrationEntries and GetNodeSelectors call, respectively. Those tests could be protected by a build tag that the datastore-mysql and datastore-postgres integration test suites build with. I know that is a bit much to bite off, so I'd be happy to help with this as a follow-up PR.

Thanks for looking into this @azdagron. I've addressed all the comments with the new updated diff. Would appreciate if you could take a look.

File: doc/plugin_server_datastore_sql.md [hunk 4]
```diff
@@ -2,17 +2,18 @@
...
-| max_open_conns    | The maximum number of open db connections (default: unlimited)             |
-| max_idle_conns    | The maximum number of idle connections in the pool (default: 2)            |
-| conn_max_lifetime | The maximum amount of time a connection may be reused (default: unlimited) |
-| disable_migration | True to disable auto-migration functionality. Use of this flag allows finer control over when datastore migrations occur and coordination of the migration of a datastore shared with a SPIRE Server cluster. Only available for databases from SPIRE Code version 0.9.0 or later. |
+| Configuration        | Description                                                                |
+| ---------------------| -------------------------------------------------------------------------- |
+| database_type        | database type                                                              |
+| connection_string    | connection string                                                          |
+| ro_connection_string | read-only connection string, used for read-only queries if set             |
```
[the remark this answers is not in the document]
Comment (kirutthika) [reply doc/plugin_server_datastore_sql.md]: Added additional section for Read Only connections.

**Comment by azdagron** [comment]
I'll take a look shortly @kirutthika!

For future reference, force-pushing after a review has taken place makes it hard to see what has changed since the last review. Don't worry about having lots of commits; it's easy to squash them before merge.

**Comment by kirutthika** [comment]
> I'll take a look shortly @kirutthika!
> 
> For future reference, force-pushing after a review has taken place makes it hard to see what has changed since the last review. Don't worry about having lots of commits; it's easy to squash them before merge.

Ah, I see. I did that for DCO checks. Will make sure the future commits are not force-pushed. Let me know if you want me to create a new diff.

**Pull request merged**